permissions/src/iosMain/kotlin/PermissionTimerHelper.kt

```diff
@@ -33,21 +33,20 @@ class PermissionTimerHelper<P:Permission>(private val permissionManager: Permiss
 
     suspend fun startMonitoring(interval: Long) {
         updateLastPermission()
-        if (timerJob != null) return
-
-        launchTimerJob(interval)
+        launchTimerJob(interval, false)
     }
 
-    private suspend fun launchTimerJob(interval: Long) {
+    private suspend fun launchTimerJob(interval: Long, isRelaunch: Boolean) {
         timerLock.withLock {
+            if (timerJob != null && !isRelaunch) return@withLock
```

Comment:
A bit more on what I meant with state:

```kotlin
sealed class TimerJobState {
	
	class TimerNotRunning():TimerJobState {
		fun startTimer()= TimerRunning()
		}
	class TimerRunning():TimerJobState {
		val job:Job: etc
		init {
			launch() {
				etc
			}
		}

		fun	stopTimer( etc ):TimerNotRunning() {
			job.cancel()
		}
    }

}
```

Then you can 

```kotlin
val timerState = TimerNotRunning()

if (timerJobState is TimerNotRunning) { timerJobState = timerJobState.startTimer() }

...

if (timerJobState is TimerRunning) { timerJobState = timerJobState.stopTimer() }
```

obviously needs correct visibility, coroutinescope, block you want to run,  etc but gives nice encapsulation with not really much more code. This kind of code seems easy, but we missed the null check (can always happen,but it shows it's not _that_ easy).